src/gantt.ts

```diff
@@ -671,104 +671,112 @@ module powerbi.extensibility.visual {
-                            name: categoryValue as string,
-                            start: startDate,
-                            end: null,
-                            duration: duration,
-                            taskType: taskType ? taskType.name : "",
-                            description: categoryValue as string,
-                            tooltipInfo: [],
-                            selected: false,
-                            identity: selectionId,
-                            extraInformation: extraInformation,
-                            daysOffList: []
-                        };
-
-                        let durationUnit: string = settings.general.durationUnit;
-                        durationUnit = (GanttDurationUnitType.indexOf(durationUnit) !== -1 && durationUnit) || "day";
-
-                        task.end = d3.time[durationUnit].offset(task.start, task.duration);
-                        if (settings.daysOff.show) {
-                            let datesDiff: number = 0;
-                            do {
-                                task.daysOffList = Gantt.calculateDaysOff(
-                                    +settings.daysOff.firstDayOfWeek,
-                                    new Date(task.start.getTime()),
-                                    new Date(task.end.getTime())
-                                );
-
-                                if (task.daysOffList.length) {
-                                    // extra duration calculating in days
-                                    let extraDuration: number = task.daysOffList
-                                        .map((item) => item[1])
-                                        .reduce((prevValue, currentValue) => prevValue + currentValue);
-
-                                    extraDuration = Gantt.transformExtraDuration(durationUnit, extraDuration);
-                                    task.end = d3.time[durationUnit].offset(task.start, task.duration + extraDuration);
-
-                                    const lastDayOff: Date = task.daysOffList[task.daysOffList.length - 1][0];
-                                    datesDiff = Math.ceil((task.end.getTime() - lastDayOff.getTime()) / MillisecondsInADay);
-                                }
-                            } while (task.daysOffList.length && datesDiff - DaysInAWeekend > DaysInAWeek);
+                const selectionId: powerbi.extensibility.ISelectionId = selectionBuider.createSelectionId();
+                const extraInformation: ExtraInformation[] = [];
+                const resource: string = (values.Resource && values.Resource[index] as string) || "";
+                const startDate: Date = (values.StartDate
+                    && Gantt.isValidDate(values.StartDate[index] as Date) && values.StartDate[index] as Date)
+                    || new Date(Date.now());
+
+                if (values.ExtraInformation && Object.keys(values.ExtraInformation).length) {
```

Comment:
Please keep Object.keys(values.ExtraInformation) to a local variable in order not to compute keys twice.

Reply:
Done